services/server/cg_server/main.py

```diff
@@ -67,12 +70,13 @@ def verify_password(username, password):
 if port := os.getenv("POSTGRES_PORT", None):
     connection_options["port"] = port
 
-conn = psycopg2.connect(**connection_options)
+conn_pool = psycopg2.pool.SimpleConnectionPool(1, 20, **connection_options)
```

Comment:
Let's have the number of connections be configurable via. an environment variable, and default to 20 if not explicitly defined